include/graphblas/reference/blas2.hpp

```diff
@@ -409,13 +412,34 @@ namespace grb {
 				}
 				// check for sparsity at source
 				if( !dense_hint ) {
-					if( !internal::getCoordinates( source_vector ).assigned( source_index ) ) {
+					if( config::PREFETCHING< reference >::enabled() ) {
+						size_t dist = k + 2 * config::PREFETCHING< reference >::distance();
+						if( dist < nz ) {
+							const size_t prefetch_target_assigned = matrix.row_index[ dist ];
+							src_coordinates.prefetch_assigned( prefetch_target_assigned );
```

Comment:
@anhansson -- this is the branch that didn't do much correct, and could be disabled? Just to check before I push a change that disables it

Reply:
Hope you had awesome holidays!! I looked at it further and this is actually not a straight index prefetch-- it's not immediately clear why this branch should not be effective. The distance of course can be off with some machine- and data-dependent factor, even when relative to the data prefetch, but that's an inherent problem with software prefetching. I'll hence merge it as-is, though will likely add some documentation before doing so~